### PR TITLE
Gradle plugin cleanup

### PR DESCRIPTION
**Pull request by shamsasari**
The Gradle plugin code has become complex and is hard to navigate. There's also code that is unnecessary, or not the "correct" way of doing things in Gradle. This clean up begins that process by making the code more manageable and easier to read. Further cleanup will probably be needed as `GradleEnclavePlugin` is still too big.

One of the main changes that has been done is making more use of Gradle task properties and providers and taking advantage of the auto-wiring of task dependencies. More info about this can be found here: https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_input_output_side_effects. A good example of this is the `EnclaveClassName` task, which now exposes an `enclaveClassName(): Provider<String>` function that will automatically wire-up the "depends on" relationship to any other task that needs the enclave class name.

There was also tasks that were either not used, or could be removed entirely.

### CODE REVIEW

**Comment by shamsasari** [comment]
@bonifazi-r3 or @ac101m can you you please review. I've marked this as draft to make sure it doesn't land before #81 .

@bonifazi-r3 what's the status of the QA tests? Do you want me to wait until they are green until we merge this?

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveBundleJar.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveBundleJar.kt:25]: Isn't this an Output ? Just asking

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveClassName.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveClassName.kt:18]: Typo, `depdendent`

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveFatJar.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/EnclaveFatJar.kt:66]: Not an expert, but don't we miss an `Output` in this? I.e. my understanding is that when we create the task, at that moment we set the output (the name in this case).

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GenerateEnclaveMetadata.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GenerateEnclaveMetadata.kt:38]: Mmm, trying to understand... Don't we miss an @Output in this as well anyway?

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt:135]: I know that sounds silly, but why don't start to we put code like this this in functions? It is polluting

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt
Comment (bonifazi-r3) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt:297]: I don't understand this provider mechanism here. Why don't we just set the outputs with a File?

**Review by bonifazi-r3 — CHANGES_REQUESTED** [review]
First round, few questions

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt
Comment (ac101m) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt:391]: Do we need this? This `when()` statement seems to be exhaustive

File: plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/SignEnclave.kt
Comment (ac101m) [inline plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/SignEnclave.kt:28]: Missed one!

(input/output in variable name)

**Review by ac101m — COMMENTED** [review]
A few minor things

**Pull request closed without merging**
